sql/src/main/java/io/crate/execution/engine/sort/InputFieldComparator.java

```diff
@@ -50,7 +50,7 @@
                          Iterable<? extends LuceneCollectorExpression<?>> collectorExpressions,
                          Input input,
                          DataType valueType,
-                         Object missingValue) {
+                         @Nullable Object missingValue) {
```

Comment:
Shouldn't we rather fix this on the call-site and make sure we pass in a non-null value? Maybe with an assertion here.

Reply:
Actually, the fix doesn't match the commit message at all? There are still null values in use.

Reply:
The NPE is avoided by not using the null value instead of failing. 

But I agree a call-site fix would be better. I'm not exactly sure what the MAX value for the string should be though (SortSymbolVisitor is the source of this entire thing and it's precisely because in case of String we don't deal with the min/max)

Reply:
Nevermind, I miss-read the commit msg